asset/decompress/zip.go

```
@@ -0,0 +1,36 @@
+	zipFile, err := d.assetService.GetFile(ctx, assetID)
+	if err != nil {
+		return err
+	}
+	defer zipFile.Close()
+
+	// Create a temporary file to store the zip content
+	// Implementation of async zip extraction
+	return nil
+}
+
+func (d *ZipDecompressor) processZipFile(ctx context.Context, zipReader *zip.Reader) error {
```

golangci-lint (1.62.2)

func `(*ZipDecompressor).processZipFile` is unused

(unused)